[PM-7231] Product Switcher within navigation sidebar

nick-livefront opened the pull request:
## Type of change

<!-- (mark with an `X`) -->

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Following the same logic of the current Product Switcher in the upper right corner, available products and/or additional products should show at the bottom of the navigation sidebar.  

## Code changes

<!--Explain the changes you've made to each file or major component. This should help the reviewer understand your changes-->
<!--Also refer to any related changes or PRs in other repositories-->

- **product-switcher.service.ts:** 
   - Moved the product switcher logic from `product-switcher-content.component.ts` into a service so it can be shared with the existing product switcher and the new one that exists in the navigation UI.
  - I did not rename the `bento` & `other` properties but I can if there is a desire for a more general term than `bento`
  - Added `navigationUIDetails` to accommodate different verbiage between the navigation product switcher and the current one. 
- **navigation-switcher.component.ts**:  
  - Component to display product offerings vertically within the navigation sidebar. 
  - The active product is always hidden   
- **toggle-width.component.ts**
  - This was absolutely positioned before this change because it was the only item at the bottom of the navigation, and probably because it is only shown in the dev environments.  I removed the absolutely positioning because it will always display below the product switcher.
- **tailwind.config.base.js**
  -  Added `xxs` font size to match the designs for the "switch" hover text
- Added product switcher to each navigation instance (if I missed one, let me know!). Also made the `nav` element a flex container so the product switcher can be pushed to the bottom of the nav.
  - `secrets-manager/layout/navigation.component.html`
  - `organizations/layouts/organization-layout.component.html`
  - `layouts/user-layout.component.html`
  - `admin-console/providers/providers-layout.component.html`
  
### Open question

- There is a `PaymentMethodWarningsComponent` that exists below my implementation of the product switcher. I wasn't able to get this to show by manipulating the code in a reasonable amount of time. Will this cause an overlap of sorts?

## Screenshots

| More from Bitwarden | Available Products |
| --------------------- | -------------------| 
|![Screen Shot 2024-04-18 at 10 52 26 AM](https://github.com/bitwarden/clients/assets/125900171/cdb5c74d-7959-45c8-b1c1-b4673f5a3cd6) | ![Screen Shot 2024-04-18 at 11 14 47 AM](https://github.com/bitwarden/clients/assets/125900171/cfe01353-6588-4c21-b755-a27d1e3e162b) |


## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

nick-livefront commented:
Converting to draft, I realized I missed some acceptance criteria. I need to add some reference events.

eliykat reviewed (changes requested):
Hi @nick-livefront!

## Question

> There is a PaymentMethodWarningsComponent that exists below my implementation of the product switcher. I wasn't able to get this to show by manipulating the code in a reasonable amount of time. Will this cause an overlap of sorts?

Even though this component is rendered by the sidebar, it actually contains a `bit-banner` which is positioned at the top of the page. Here is it is with your changes:

![Screenshot 2024-04-25 at 2 12 49 PM](https://github.com/bitwarden/clients/assets/31796059/c0f7b25a-d633-454c-a348-3ebd10ec27bc)

## Other comments

Can you please add a Storybook story for the navigation product switcher? See the existing stories for the current [product switcher](https://components.bitwarden.com/?path=/docs/web-product-switcher--docs) and the [layout](https://components.bitwarden.com/?path=/docs/component-library-layout--docs), for reference.

The AC Team changes are simple and otherwise approved, however even with multiple teams involved in this PR review, we have no code owners for the switcher itself. I'll check internally who should be responsible for it.

nick-livefront commented:
> Even though this component is rendered by the sidebar, it actually contains a `bit-banner` which is positioned at the top of the page. 

@eliykat Thank you for validating and the clarification! 

> Can you please add a Storybook story for the navigation product switcher? See the existing stories for the current [product switcher](https://components.bitwarden.com/?path=/docs/web-product-switcher--docs) and the [layout](https://components.bitwarden.com/?path=/docs/component-library-layout--docs), for reference.

Yes! I added new stories for the navigation switcher and fixed the existing product switcher stories which I broke with the added service.

[ad8bb93](https://github.com/bitwarden/clients/pull/8810/commits/ad8bb930c5a7d3e564bf972ea7a68ada296d1e66)

eliykat commented:
@bitwarden/team-platform-dev should own the layout components, see #8814. I've requested their review, please treat them as a code owner (i.e. don't merge without their approval).

Hinton commented on apps/web/src/app/layouts/product-switcher/navigation-switcher/navigation-switcher.component.html:
```diff
@@ -0,0 +1,37 @@
+    class="tw-group"
+  >
+    <ng-container slot="end">
+      <span class="tw-text-xxs tw-hidden group-hover:tw-block group-focus:tw-block">
+        {{ "switch" | i18n }}
+      </span>
```
**question:** In figma this seems to be the `xs` not `xss`?

nick-livefront replied:
Can you link to where you're seeing that? 

In this [mockup](https://www.figma.com/file/I0kvnK2cITmfrRRzl5lATx/Product-Switcher-Iterations?type=design&node-id=1111-14848&mode=design&t=yY9Ovb2rYd1QMwWB-4), `10px font / 14px line height` is set.

<img width="548" alt="Screenshot 2024-04-26 at 1 29 49 PM" src="https://github.com/bitwarden/clients/assets/125900171/4b2998d0-618b-4929-a0bf-aec73e6ce38d">

Hinton replied:
Apologies, I think I confused the sections in the Figma UI. 10p seems right.

Hinton commented on apps/web/src/app/layouts/product-switcher/navigation-switcher/navigation-switcher.component.html:
```diff
@@ -0,0 +1,37 @@
+    [hideActiveStyles]="true"
+    class="tw-group"
+  >
+    <ng-container slot="end">
```
**issue:** The end slot is not great for this use case, as it's not clickable. It's quite confusing to have the text "Switch" appear and not have the area be clickable. It's also not aligned with the regular text since it's in a different element.

I wonder if we can lift it into the regular text, however the space in this area is a bit cramped already.

nick-livefront replied:
Good point, this appears to be a new pattern for the `NavItemComponent` to handle. Other instances that I'm seeing have content that is interact-able within the end slot which covers this case.  Setting it a part of the text would throw off the spacing a little bit.

Is a path forward a question back to design about the implications of having text shown like this?

Hinton replied:
I've asked internally, I think code wise this seems reasonable for now and we can revisit it presuming we have a plan. I'm concerned about the non clickable text though since I expect the click text not being clickable will be quite confusing for the users.

Hinton commented on apps/web/src/app/layouts/product-switcher/navigation-switcher/navigation-switcher.component.html:
```diff
@@ -0,0 +1,37 @@
+    </ng-container>
+  </bit-nav-item>
+  <section
+    *ngIf="((moreProducts$ | async) ?? []).length > 0"
```
**issue:** This will setup two listeners for the observable. It's generally better to wrap it in a ng-content. This would also let you. simplify the section if condition.

nick-livefront replied:
A solution with `ng-content` is escaping me, I ended up with an `ng-container`: [6dd95d6](https://github.com/bitwarden/clients/pull/8810/commits/6dd95d60f6003e1b56033fb419606d34cc1558f3). Feel free to point me in the right direction if there is a simpler way!

Hinton replied:
Apologies, I always confuse `ng-content` and `ng-container`. Container is correct and this looks good!

Hinton commented on apps/web/src/app/layouts/product-switcher/navigation-switcher/navigation-switcher.component.html:
```diff
@@ -0,0 +1,37 @@
+    class="tw-mt-2 tw-flex tw-w-full tw-flex-col tw-gap-2 tw-border-0 tw-border-t tw-border-solid tw-border-t-text-alt2"
+  >
+    <span class="tw-text-xs !tw-text-alt2 tw-p-2 tw-pb-0">{{ "moreFromBitwarden" | i18n }}</span>
+    <a
+      *ngFor="let more of moreProducts$ | async"
+      [href]="more.marketingRoute"
+      target="_blank"
+      rel="noreferrer"
+      class="tw-flex tw-py-2 tw-px-4 tw-font-semibold !tw-text-alt2 !tw-no-underline hover:tw-bg-primary-300/60 [&>:not(.bwi)]:hover:tw-underline"
+    >
+      <i class="bwi bwi-fw {{ more.icon }} tw-mt-1"></i>
+      <div>
+        {{ more.navigationUIDetails?.name ?? more.name }}
+        <div *ngIf="more.navigationUIDetails?.supportingText" class="tw-text-xs tw-font-normal">
+          {{ more.navigationUIDetails.supportingText }}
+        </div>
+      </div>
+    </a>
```
**question**: This duplicates much of the styling from `bit-nav-item`, is there any possibility of re-using the component here to prevent needing to update multiple places?

I suspect we're missing support for the supporting text?

nick-livefront replied:
> I suspect we're missing support for the supporting text?

That was my conclusion as well. Maybe a solution for both this and the [end slot comment](https://github.com/bitwarden/clients/pull/8810#discussion_r1581176874) could be to make a dedicated component for this implementation. Con would be duplicated styles between the `bit-nav-item` and the new component but it could solve both issues here?

Hinton commented on apps/web/src/app/layouts/product-switcher/navigation-switcher/navigation-switcher.stories.ts:
**issue:** This story is written using the old now deprecated storybook format. Could we update it to the modern approach? A good example is `button.stories.ts`.

nick-livefront replied:
Thanks for pointing that out, I updated to CSF3: [dafaf01](https://github.com/bitwarden/clients/pull/8810/commits/dafaf01b395851cf42f537acf76193d4b0109fbc)

Hinton commented on apps/web/src/app/layouts/product-switcher/navigation-switcher/navigation-switcher.stories.ts:
```diff
@@ -0,0 +1,154 @@
+
+import { NavigationProductSwitcherComponent } from "./navigation-switcher.component";
+
+@Directive({
+  selector: "[mockOrgs]",
+})
+class MockOrganizationService implements Partial<OrganizationService> {
+  private static _orgs = new BehaviorSubject<Organization[]>([]);
+  organizations$ = MockOrganizationService._orgs; // eslint-disable-line rxjs/no-exposed-subjects
+
+  @Input()
+  set mockOrgs(orgs: Organization[]) {
+    this.organizations$.next(orgs);
+  }
+}
+
+@Directive({
+  selector: "[mockProviders]",
+})
+class MockProviderService implements Partial<ProviderService> {
+  private static _providers = new BehaviorSubject<Provider[]>([]);
+
+  async getAll() {
+    return await firstValueFrom(MockProviderService._providers);
+  }
+
+  @Input()
+  set mockProviders(providers: Provider[]) {
+    MockProviderService._providers.next(providers);
+  }
+}
```
**question:** Now that we have a proper product switcher service, should we mock that instead? We're not particularly interested in testing the behaviour of the service in stories.

nick-livefront replied:
In most cases I would agree that the underlying implementation shouldn't be necessary. In this case, mocking the `ProductSwitcherService` would result in consumers of Storybook to see instances of `ProductSwitcherItem` being set. This would be inaccurate to the implementation of the component because a developer can't change the items, only the underlying data drives the UI permutations.

In my opinion it is more beneficial for a developer to see what is driving the different UI states. And this way the Storybook will always be up to date with the implementation.

Hinton commented on apps/web/src/app/layouts/product-switcher/navigation-switcher/navigation-switcher.stories.ts:
```diff
@@ -0,0 +1,154 @@
+  props: args,
+  template: `
+    <router-outlet [mockOrgs]="mockOrgs" [mockProviders]="mockProviders"></router-outlet>
+    <bit-layout>
+      <nav slot="sidebar" class="tw-flex tw-flex-col tw-h-full">
+        <navigation-product-switcher></navigation-product-switcher>
+      </nav>
+    </bit-layout>
```
**question**: Do we want just use a div with background with with instead of bit-layout? It would result in less scrolling when viewing the storybook.

nick-livefront replied:
Feels like an improvement to me: [7c622e8](https://github.com/bitwarden/clients/pull/8810/commits/7c622e8e59aacdf646246df2e6dfefb55451ca55)

Hinton commented on apps/web/src/app/layouts/product-switcher/shared/product-switcher.service.ts:
```diff
@@ -0,0 +1,158 @@
+          appRoute: "/vault",
+          marketingRoute: "https://bitwarden.com/products/personal/",
+          isActive:
+            !this.router.url.includes("/sm/") &&
+            !this.router.url.includes("/organizations/") &&
+            !this.router.url.includes("/providers/"),
+        },
+        sm: {
+          name: "Secrets Manager",
+          icon: "bwi-cli",
+          appRoute: ["/sm", smOrg?.id],
+          marketingRoute: "https://bitwarden.com/products/secrets-manager/",
+          isActive: this.router.url.includes("/sm/"),
+          navigationUIDetails: {
+            supportingText: this.i18n.transform("secureYourInfrastructure"),
+          },
+        },
+        ac: {
+          name: "Admin Console",
+          icon: "bwi-business",
+          appRoute: ["/organizations", acOrg?.id],
+          marketingRoute: "https://bitwarden.com/products/business/",
+          isActive: this.router.url.includes("/organizations/"),
+        },
+        provider: {
+          name: "Provider Portal",
+          icon: "bwi-provider",
+          appRoute: ["/providers", providers[0]?.id],
+          isActive: this.router.url.includes("/providers/"),
+        },
+        orgs: {
+          name: "Organizations",
+          icon: "bwi-business",
+          marketingRoute: "https://bitwarden.com/products/business/",
+          navigationUIDetails: {
+            name: "Share your passwords",
+            supportingText: this.i18n.transform("protectYourFamilyOrBusiness"),
+          },
+        },
+      };
```
We are using TS 4.9 now, could you tweak this to:

```suggestion
      const products = {
        pm: {
          name: "Password Manager",
          icon: "bwi-lock",
          appRoute: "/vault",
          marketingRoute: "https://bitwarden.com/products/personal/",
          isActive:
            !this.router.url.includes("/sm/") &&
            !this.router.url.includes("/organizations/") &&
            !this.router.url.includes("/providers/"),
        },
        sm: {
          name: "Secrets Manager",
          icon: "bwi-cli",
          appRoute: ["/sm", smOrg?.id],
          marketingRoute: "https://bitwarden.com/products/secrets-manager/",
          isActive: this.router.url.includes("/sm/"),
          navigationUIDetails: {
            supportingText: this.i18n.transform("secureYourInfrastructure"),
          },
        },
        ac: {
          name: "Admin Console",
          icon: "bwi-business",
          appRoute: ["/organizations", acOrg?.id],
          marketingRoute: "https://bitwarden.com/products/business/",
          isActive: this.router.url.includes("/organizations/"),
        },
        provider: {
          name: "Provider Portal",
          icon: "bwi-provider",
          appRoute: ["/providers", providers[0]?.id],
          isActive: this.router.url.includes("/providers/"),
        },
        orgs: {
          name: "Organizations",
          icon: "bwi-business",
          marketingRoute: "https://bitwarden.com/products/business/",
          navigationUIDetails: {
            name: "Share your passwords",
            supportingText: this.i18n.transform("protectYourFamilyOrBusiness"),
          },
        },
      } satisfies Record<string, ProductSwitcherItem>;
```

nick-livefront replied:
Nice catch! [bd54d87](https://github.com/bitwarden/clients/pull/8810/commits/bd54d8713dd2c715d01d30f057d8736667e90917)

Hinton commented on apps/web/src/app/layouts/product-switcher/shared/product-switcher.service.ts:
```diff
@@ -0,0 +1,158 @@
+   */
+  isActive?: boolean;
+
+  /**
+   * A product switcher item can be shown in the left navigation menu
+   * with different content than the main product switcher.
+   */
+  navigationUIDetails?: {
+    /** Alternative navigation menu name */
+    name?: string;
+    /** Supporting text that is shown in the "more from bitwarden" section */
+    supportingText?: string;
+  };
```
I think the term `navigationUI` is confusing since technically both the top and sidebar are navigation sections. We already use the term `sidebar` in other places, maybe calling this `sidebarOverrides` might be appropriate. We're overriding the name, although there is no concept of supporting text in the other switcher 🤔. It's also related to the marketing since it only appears in the `otherProducts` section.

nick-livefront replied:
Maybe I went a step further, I ended up as "otherProductOverrides" and updated the comment to hopefully be more specific. Let me know what you think! Obligatory, naming is hard 😄 

[749fcdf](https://github.com/bitwarden/clients/pull/8810/commits/749fcdfb338bb25258ac6af45eb38db3e1326c72)

Hinton replied:
Seems reasonable naming, thanks.

Hinton commented on apps/web/src/app/layouts/product-switcher/shared/product-switcher.service.ts:
```diff
@@ -0,0 +1,158 @@
+  navigationUIDetails?: {
+    /** Alternative navigation menu name */
+    name?: string;
+    /** Supporting text that is shown in the "more from bitwarden" section */
```
**issue:** We call this `otherProducts$ section` above.

nick-livefront replied:
👍 addressed as a part of the attribute rename: [749fcdf](https://github.com/bitwarden/clients/pull/8810/commits/749fcdfb338bb25258ac6af45eb38db3e1326c72)

Hinton commented on apps/web/src/app/layouts/product-switcher/navigation-switcher/navigation-switcher.component.ts:
```diff
@@ -0,0 +1,24 @@
+  templateUrl: "./navigation-switcher.component.html",
+})
+export class NavigationProductSwitcherComponent {
+  // Use margin-top: auto to push the component to the bottom of the parent flex container
+  @HostBinding("style.margin-top") marginTop = "auto";
```
**suggestion:** I don't have a to strong preference, but I feel like using a host binding here a bit deceptive. It's not easily understood from looking at the layouts that a margin top will be applied. I think lifting this into the various layouts would make it more clear even though we duplicate some content.

We should also use a tailwind class instead of style.

nick-livefront replied:
I appreciate the perspective! It is slightly implementation specific, theoretically there could be an implementation where the margin top isn't needed.

I added it as a tailwind class on each implementation: [6e49eea](https://github.com/bitwarden/clients/pull/8810/commits/6e49eeaf09ec567e01ea796e364a338cdcbe2edf)

Hinton commented on apps/web/src/app/layouts/product-switcher/navigation-switcher/navigation-switcher.component.ts:
```diff
@@ -0,0 +1,24 @@
+
+  constructor(private productSwitcherService: ProductSwitcherService) {}
+
+  accessibleProducts$: Observable<ProductSwitcherItem[]> =
+    this.productSwitcherService.products$.pipe(
+      map((products) => (products.bento ?? []).filter((item) => !item.isActive)),
+    );
+
+  moreProducts$: Observable<ProductSwitcherItem[]> = this.productSwitcherService.products$.pipe(
+    map((products) => (products.other ?? []).filter((item) => !item.isActive)),
+  );
```
**suggestion**: Let's change these to proteted readonly to denote they won't change.
```suggestion
  protected readonly accessibleProducts$: Observable<ProductSwitcherItem[]> =
    this.productSwitcherService.products$.pipe(
      map((products) => (products.bento ?? []).filter((item) => !item.isActive)),
    );

  protected readonly moreProducts$: Observable<ProductSwitcherItem[]> = this.productSwitcherService.products$.pipe(
    map((products) => (products.other ?? []).filter((item) => !item.isActive)),
  );
```

nick-livefront replied:
👍  [68792b8](https://github.com/bitwarden/clients/pull/8810/commits/68792b81b3982696d6963175c33cdcddc0d75317)

Hinton reviewed (changes requested):
Hey @nick-livefront,

Looks generally good, I found a handful of minor changes related to our best practices that we attempt to apply to new code. Other than that it would be nice to update the storybook to the new format since the old one is deprecated, and an open question about possibly using bit-nav-item for some duplicated styling.

danielleflinn commented on libs/components/tailwind.config.base.js:
```diff
@@ -86,6 +86,9 @@ module.exports = {
     }),
     extend: {
+      fontSize: {
+        xxs: ["0.625rem", "0.875rem"],
```
I'll leave the specifics of how this question relates to this PR to @jtouchstonebw but its worth mentioning that in the new DS design the required indicator on fields is a similar size

nick-livefront commented on apps/web/src/app/layouts/product-switcher/product-switcher-content.component.ts:
```diff
@@ -44,99 +12,9 @@ export class ProductSwitcherContentComponent {
-  ) {}
+  get products$() {
+    return this.productSwitcherService.products$;
+  }
```
[7714118](https://github.com/bitwarden/clients/pull/8810/commits/7714118116ba4b65edfd30ce389a07a383759000)

nick-livefront commented on apps/web/src/app/layouts/toggle-width.component.ts:
```diff
@@ -10,7 +10,7 @@ import { NavigationModule } from "@bitwarden/components";
     icon="bwi-bug"
     *ngIf="isDev"
-    class="tw-absolute tw-bottom-0 tw-w-full"
+    class="tw-w-full"
```
[5a5f9ee](https://github.com/bitwarden/clients/pull/8810/commits/5a5f9ee1477ca5df30932621e405eb0d725be15a)

Hinton commented:
Hey @nick-livefront,

Sorry for the delay. I've discussed the UX with design and we decided to remove the `Switch` text completely. This should resolve the concerns about it not being clickable. We also discussed some concerns about items changing order as you change product, and the desire is to keep the active product in the nav with a `active` state similar to general nav items.

Does this seem reasonable to you?

I also had a chat with @willmartian about the duplicated styling and we decided it's fine as it is. There are some future tasks to change how the nav works which would resolve this.

nick-livefront commented:
> Sorry for the delay. I've discussed the UX with design and we decided to remove the `Switch` text completely. This should resolve the concerns about it not being clickable. We also discussed some concerns about items changing order as you change product, and the desire is to keep the active product in the nav with a `active` state similar to general nav items.
> 
> Does this seem reasonable to you?

@Hinton I appreciate you tracking down these answers and they both make sense to me! I'll also note them in the ticket.

- Removed "switch" text [fc39266](https://github.com/bitwarden/clients/pull/8810/commits/fc3926645813d55237164a7f933c1cac65c4c7b0)
- Allow for `NavItemComponent` to show active styles [ac9ca1c](https://github.com/bitwarden/clients/pull/8810/commits/ac9ca1cba0e7b0c668d8b1a0f0445570c5e8864d)

Hinton commented on bitwarden_license/bit-web/src/app/admin-console/providers/providers.module.ts:
```diff
@@ -45,6 +46,7 @@ import { SetupComponent } from "./setup/setup.component";
     PaymentMethodWarningsModule,
     TaxInfoComponent,
     DangerZoneComponent,
+    RouterModule,
```
**issue:** I don't believe this is required?

nick-livefront replied:
🤔 You are correct. During development I wasn't seeing the `active` state within the Admin Console, a little research pointed to this being the issue. But removing it now I'm seeing everything as it should.  [b7ebcac](https://github.com/bitwarden/clients/pull/8810/commits/b7ebcac96879752298ba23c3a3644697134bba77)

Hinton commented on apps/web/src/app/layouts/product-switcher/navigation-switcher/navigation-switcher.stories.ts:
```diff
@@ -0,0 +1,172 @@
+  moreFromBitwarden: "More from Bitwarden",
+  secureYourInfrastructure: "Secure your infrastructure",
+  protectYourFamilyOrBusiness: "Protect your family or business",
+  switch: "Switch",
```
**issue:** We can remove `switch` from the story.

nick-livefront replied:
👍 [b80c212](https://github.com/bitwarden/clients/pull/8810/commits/b80c2120ad04cd85c04ac8c8272c45bf6d327dfc)

Hinton commented on apps/web/src/app/layouts/product-switcher/navigation-switcher/navigation-switcher.component.html:
```diff
@@ -0,0 +1,32 @@
+    [route]="product.appRoute"
+  >
+  </bit-nav-item>
+  <ng-container *ngIf="(moreProducts$ | async) ?? [] as moreProducts">
```
**question:** Are we overly defensive here? The service can't return null, so `?? []` should never trigger. Especially since it's assigned a default value in the typescript code.

Hinton reviewed (commented):
Two minor tweaks and this should be good from my perspective. i left one open ended question to.

LRNcardozoWDF reviewed (dismissed):
Nice job! LGTM

Thomas-Avery reviewed (approved):
SM changes look good, thank you.

Hinton reviewed (dismissed):
LGTM, thanks for iterating on this @nick-livefront.

LRNcardozoWDF reviewed (dismissed):
Nice job, LGTM!

Hinton commented on bitwarden_license/bit-web/src/app/admin-console/providers/providers-layout.component.html:
```diff
@@ -40,6 +40,11 @@
       route="settings"
       *ngIf="showSettingsTab(provider)"
     ></bit-nav-item>
+    <<<<<<< HEAD
```
Missed diff

nick-livefront replied:
👍 [c8ca9f9](https://github.com/bitwarden/clients/pull/8810/commits/c8ca9f9054a6fdf2e011d7b708aa3f0e335c78e5)

eliykat reviewed (dismissed):
Please note UI tests also need to be reviewed/approved

nick-livefront commented:
@Hinton @eliykat @LRNcardozoWDF @Thomas-Avery 

I had 3 defects come back, I added each as a commit here. Let me know if an alternate process is preferred!

[PM-7899](https://bitwarden.atlassian.net/browse/PM-7899) - [c78490b](https://github.com/bitwarden/clients/pull/8810/commits/c78490b665ee676c3240872edaf054986e4c42da)
[PM-7951](https://bitwarden.atlassian.net/browse/PM-7951) - [0b06f42](https://github.com/bitwarden/clients/pull/8810/commits/0b06f427a332f06118fdee340b57bf5aa5800cbe)
[PM-7984](https://bitwarden.atlassian.net/browse/PM-7948) - [a12b95c](https://github.com/bitwarden/clients/pull/8810/commits/a12b95c7bdbbac81567fad2864afe5dd75a77699)

eliykat reviewed (approved):
Rubber stamping as no impact on AC-team owned files.

eliykat commented:
I assume you mean https://github.com/bitwarden/clients/pull/9128? :)

nick-livefront commented:
> I assume you mean #9128? :)

Yes I do! Thank you 🙇

nick-livefront commented:
@Hinton @eliykat @LRNcardozoWDF @Thomas-Avery

https://github.com/bitwarden/clients/pull/9128 merged into this branch! Ready for review, I appreciate the patience 🙌

Hinton commented:
@nick-livefront @jtouchstonebw mentioned in [chromatic](https://www.chromatic.com/pullrequest?appId=62a88a6de5b807fa98886113&number=8810&activeElementId=comparison-test-key-662a75e65ee9411d85cb55b6-1200px&view=changes) that spacing between icon and text looks a bit smaller than in general.

nick-livefront commented:
> @nick-livefront @jtouchstonebw mentioned in [chromatic](https://www.chromatic.com/pullrequest?appId=62a88a6de5b807fa98886113&number=8810&activeElementId=comparison-test-key-662a75e65ee9411d85cb55b6-1200px&view=changes) that spacing between icon and text looks a bit smaller than in general.

@Hinton @jtouchstonebw I missed some margin on the icons in that section, [8d04566](https://github.com/bitwarden/clients/pull/8810/commits/8d04566b0272d4703b444754a0bb319ae26e25cc) adds the margin to match other icons in the navigation menu
